pkg/apis/serving/v1alpha1/route_validation.go

```diff
@@ -19,15 +19,18 @@ package v1alpha1
 import (
 	"context"
 	"fmt"
+	"strconv"
 
 	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/networking/pkg/apis/networking"
 	"knative.dev/pkg/apis"
 	"knative.dev/serving/pkg/apis/serving"
 )
 
 func (r *Route) Validate(ctx context.Context) *apis.FieldError {
 	errs := serving.ValidateObjectMetadata(ctx, r.GetObjectMeta()).ViaField("metadata")
-	errs = errs.Also(r.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec"))
+	errs = errs.Also(r.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec")).
+		Also(validateAnnotations(r.GetAnnotations()))
```

Comment:
We may want to put the `validateAnnotations` function into metadta_validation file https://github.com/knative/serving/blob/84baf2df32adb9732a3308edbbf1933ae7dd5aed/pkg/apis/serving/metadata_validation.go#L49

Reply:
I didn't do this because this annotation validation is intended for networking resources only; other scopes already have this kind of validation.

The intent is for the logic to be moved into knative/networking as part of knative/networking#123. I can move it there now if we think that's better.

Reply:
that probably would make more sense.

Reply:
SGTM. thanks for the explanation. Could you please add a TODO here?